### ISSUE TITLE
rank-based alternative to traceplot

### ISSUE THREAD

**Issue by tjmahr** [stan-dev/bayesplot#178]
As described by Dan here https://statmodeling.stat.columbia.edu/2019/03/19/maybe-its-time-to-let-the-old-ways-die-or-we-broke-r-hat-so-now-we-have-to-fix-it/

If we take the input into mcmc_trace(), we know the chain id and values of the iterations. Then we can compute ranks with dplyr::row_number() and facet by by chain id. 

Sanity check: Plot should be perfectly uniform when mode has just one chain.

**Comment by avehtari** [comment]
@tjmahr's comments on rank plot

- If we take the input into mcmc_trace(), we know the chain id and values of the iterations. Then we can compute ranks with dplyr::row_number() and facet by chain id. 
- Sanity check: Plot should be perfectly uniform when model has just one chain. Edit (Aki): The model should never have just one chain.

**Comment by avehtari** [comment]
@tjmahr, @paul-buerkner, @jgabry suggestions for the function names?

**Comment by paul-buerkner** [comment]
I think we could also use the `mcmc_` prefix for the ESS plots, something like:

```
mcmc_ess_local
mcmc_ess_quantile
mcmc_ess_change
```

The rank plots could simply be named `mcmc_rank` or similar.

**Comment by jgabry** [comment]
I've been traveling and am just catching up on this. I like the function names suggested by @paul-buerkner

**Comment by jgabry** [comment]
Related to this issue, we'll also need to update the bayesplot MCMC diagnostics vignette.

**Comment by avehtari** [comment]
Added to todo list: mcmc_ecdfd_r_scale for ECDF difference plots (see Fig. 14c in SBC paper)

**Comment by avehtari** [comment]
example of mcmc_ecdfd_r_scale plot
https://twitter.com/avehtari/status/1122940123522502657

**Comment by jgabry** [comment]
Thanks Aki

**Comment by ASKurz** [comment]
Would it be possible to add a `facet_args` argument to `mcmc_rank_overlay()`?

**Comment by jgabry** [comment]
Yeah I actually recently merged a PR from @hhau that added this (https://github.com/stan-dev/bayesplot/pull/221), so it should be available now if you install bayesplot from GitHub.

**Comment by ASKurz** [comment]
Excellent! Thanks for the heads up.

**Comment by timwaterhouse** [comment]
Hi all. It looks like there hasn't been any activity here for a while. Is there still interest in pursuing the remaining plot functions from this issue?

**Comment by avehtari** [comment]
ECDF difference plots were added by #282
ess-plots are still missing, but it would be nice to have them

**Comment by timwaterhouse** [comment]
Would a PR be welcome for those ESS plots that essentially collects the relevant code from @avehtari's reference code (along with tests etc)?  Are there some complexities about bayesplot that I'd need to be aware of to make this work?

**Comment by avehtari** [comment]
>  Would a PR be welcome

Yes! You can make a PR, and Jonah can then provide feedback to take into account the bayesplot requirements

**Comment by ishaan-arora-1** [comment]
hey @jgabry what do you think about implementing mcmc_ess_local, mcmc_ess_quantile, mcmc_ess_change as a part of the project

**Comment by jgabry** [comment]
Sure, sounds good to me. @avehtari what do you think? Do you have other priorities for the GSOC project or would you be happy if these are included?
